state/store.go

```diff
@@ -73,6 +79,20 @@ type Store interface {
 	Bootstrap(State) error
 	// PruneStates takes the height from which to start pruning and which height stop at
 	PruneStates(int64, int64, int64) error
+	// PruneABCIResponses will prune all ABCI responses below the given height.
+	PruneABCIResponses(int64) (uint64, error)
+	// SaveApplicationRetainHeight persists the application retain height from the application
+	SaveApplicationRetainHeight(height int64) error
+	// GetApplicationRetainHeight returns the retain height set by the application
+	GetApplicationRetainHeight() (int64, error)
+	// SaveDataCompanionRetainHeight saves the retain height set by the data companion
+	SaveDataCompanionRetainHeight(height int64) error
```

Comment:
Should probably be called `SaveCompanionBlockRetainHeight` to make it clear that this only influences block retention. (I dropped the `Data` part because otherwise the function name would be crazy long)